src/react/components/pages/projectSettings/projectSettingsPage.tsx

```diff
@@ -67,21 +73,20 @@ export default class ProjectSettingsPage extends React.Component<IProjectSetting
                     <ProjectForm
                         project={this.props.project}
                         connections={this.props.connections}
+                        appSettings={this.props.appSettings}
                         onSubmit={this.onFormSubmit}
                         onCancel={this.onFormCancel} />
                 </div>
             </div>
         );
     }
 
-    private onFormSubmit = async (formData) => {
-        const projectToUpdate: IProject = {
-            ...formData,
-        };
+    private onFormSubmit = async (project: IProject) => {
+        const isNew = !(!!project.id);
```

Comment:
is this common practice?  this looks like code smell to me

Reply:
The `!!` operator in javascript will evaluate any javascript expression and return a truthy or falsey value.  If a project is new, then the `id` property will not be set at this stage - it is handled in the project service.  

`!!project.id` will evaluate to `false` on a new project, therefore `isNew === false`.  We then use this value below to handle which page the user is transitioned to next.

I agree all the exclamation points can make this more difficult to expand... I'd be fine with inverting the condition to `const isExisting = !!project.id`.  Then it may be easier to understand.